### PR TITLE
use-operation-options-from-core-client-rest

### PR DESCRIPTION
**Pull request by qiaozha**
fixes https://github.com/Azure/autorest.typescript/issues/1902 
This PR is meant to reflect the changes here https://github.com/Azure/azure-sdk-for-js/pull/26331/commits/cb2567c6a19744aa0710b6c21c12f480ea6edf5f 

Pending on a newer version of @azure-rest/core-client release.

/cc @xirzec

### CODE REVIEW

**Review by xirzec — COMMENTED** [review]
Change looks good to me. The only other thing I'd expect to see is an update to the minimum version of `@azure-rest/core-client` in the generated package.json file.

**Comment by qiaozha** [comment]
> Change looks good to me. The only other thing I'd expect to see is an update to the minimum version of `@azure-rest/core-client` in the generated package.json file.

Thanks for the reminder, I was thinking to update the version once the @azure-rest/core-client get released.

File: packages/autorest.typescript/package.json [hunk 1]
```diff
@@ -55,7 +55,7 @@
     "@autorest/codemodel": "~4.19.2",
     "@autorest/extension-base": "^3.5.0",
     "@autorest/testmodeler": "^2.4.0",
-    "@azure-rest/core-client": "^1.1.3",
+    "@azure-rest/core-client": "1.1.4-alpha.20230630.1",
```
Comment (qiaozha) [inline packages/autorest.typescript/package.json:58]: will change this back to `"@azure-rest/core-client": "^1.1.4"` once it's released.,

**Review by xirzec — APPROVED** [review]
Looks great!

**Review by MaryGao — APPROVED** [review]
LGTM

**Pull request merged**